diff -r 0b3902dbe274 sys/Makefile
--- a/sys/Makefile	Sun Sep 13 09:15:02 2015 +0530
+++ b/sys/Makefile	Sun Feb 28 11:43:32 2016 +0530
@@ -5,6 +5,7 @@
 SUBDIR=	altq arch compat dev fs miscfs \
 	net net80211 netatalk netbt netipsec netinet netinet6 \
         netisdn netkey netmpls netnatm netsmb \
+	newdrv \
 	nfs opencrypto sys ufs uvm
 
 # interrupt implementation depends on the kernel within the port
diff -r 0b3902dbe274 sys/conf/files
--- a/sys/conf/files	Sun Sep 13 09:15:02 2015 +0530
+++ b/sys/conf/files	Sun Feb 28 11:43:32 2016 +0530
@@ -1565,3 +1565,8 @@
 # athn
 #
 include "dev/ic/files.athn"
+
+# newdrv
+defpseudo	newdrv
+file	newdrv/newdrv.c	newdrv
+#include "newdrv/files.newdrv"
diff -r 0b3902dbe274 sys/modules/Makefile
--- a/sys/modules/Makefile	Sun Sep 13 09:15:02 2015 +0530
+++ b/sys/modules/Makefile	Sun Feb 28 11:43:32 2016 +0530
@@ -112,6 +112,7 @@
 SUBDIR+=	vnd
 SUBDIR+=	zlib
 SUBDIR+=	tprof
+SUBDIR+=	newdrv
 .if (defined(NOTYET))
 SUBDIR+=	hifn		# Builds on architectures with PCI bus
 SUBDIR+=	unionfs
diff -r 0b3902dbe274 sys/modules/newdrv/Makefile
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sys/modules/newdrv/Makefile	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,10 @@
+# $NetBSD$
+
+.include "../Makefile.inc"
+
+.PATH:	${S}/newdrv
+
+KMOD=	newdrv
+SRCS=	newdrv.c
+
+.include <bsd.kmodule.mk>
diff -r 0b3902dbe274 sys/newdrv/Makefile
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sys/newdrv/Makefile	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,7 @@
+#	$NetBSD$
+
+INCSDIR= /usr/include/
+
+INCS=	newdrv.h
+
+.include <bsd.kinc.mk>
diff -r 0b3902dbe274 sys/newdrv/files.newdrv
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sys/newdrv/files.newdrv	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,4 @@
+#	$NetBSD$
+
+defpseudo	newdrv
+file	newdrv/newdrv.c	newdrv
\ No newline at end of file
diff -r 0b3902dbe274 sys/newdrv/newdrv.c
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sys/newdrv/newdrv.c	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,15 @@
+/* $NetBSD$ */
+
+/*
+ * Copyright FOSSmeet 2016
+ * Written by Random hacker <hacker@foo.bar>
+ *
+ */
+
+#include "newdrv.h"
+
+int
+sum(int a, int b)
+{
+	return a + b;
+}
diff -r 0b3902dbe274 sys/newdrv/newdrv.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sys/newdrv/newdrv.h	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,2 @@
+/* $NetBSD$ */
+int sum(int, int);
diff -r 0b3902dbe274 tests/sys/newdrv/Makefile
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/tests/sys/newdrv/Makefile	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,15 @@
+# $NetBSD$
+#
+WARNS?=6
+.include <bsd.own.mk>
+
+TESTSDIR=	${TESTSBASE}/sys/newdrv
+CPPFLAGS+=	-I${NETBSDSRCDIR}/sys -D_TEST
+
+# Depend on the kernel source files too
+DPSRCS=${NETBSDSRCDIR}/sys/newdrv/newdrv.[ch]
+
+TESTS_C=	t_newdrv
+
+.include <bsd.dep.mk>
+.include <bsd.test.mk>
diff -r 0b3902dbe274 tests/sys/newdrv/t_newdrv.c
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/tests/sys/newdrv/t_newdrv.c	Sun Feb 28 11:43:32 2016 +0530
@@ -0,0 +1,23 @@
+/* $NetBSD$ */
+
+#include <stdio.h>
+
+#include "newdrv/newdrv.c"
+#include <atf-c.h>
+
+ATF_TC(sum);
+ATF_TC_HEAD(sum, tc)
+{
+	atf_tc_set_md_var(tc, "descr", "Sum of two numbers");
+}
+ATF_TC_BODY(sum, tc)
+{
+	ATF_CHECK(sum(4, 5) == 9);
+	printf("sum of 4 + 5 == %f\n", sum(4, 5));
+}
+
+ATF_TP_ADD_TCS(tp)
+{
+	ATF_TP_ADD_TC(tp, sum);
+	return atf_no_error();
+}
